lib/Conversion/TorchOnnxToTorch/OnnxLstmExpander.cpp

```diff
@@ -0,0 +1,524 @@
+  Value f_h = b.create<AtenLinearOp>(hTy, H_prev, weights.R_f, weights.Rb_f);
+  Value f = b.create<AtenAddTensorOp>(hTy, f_x, f_h, cstOne);
+  Value f_act = createActivationByName(b, activations.f, f);
+
+  Value ct_x = b.create<AtenLinearOp>(hTy, Xt, weights.W_c, weights.Wb_c);
+  Value ct_h = b.create<AtenLinearOp>(hTy, H_prev, weights.R_c, weights.Rb_c);
+  Value ct = b.create<AtenAddTensorOp>(hTy, ct_x, ct_h, cstOne);
+  Value ct_act = createActivationByName(b, activations.g, ct);
+
+  Value C_forget = b.create<AtenMulTensorOp>(hTy, f_act, C_prev);
+  Value C_input = b.create<AtenMulTensorOp>(hTy, i_act, ct_act);
+
+  LstmCellState newCellState;
+  newCellState.C = b.create<AtenAddTensorOp>(hTy, C_forget, C_input, cstOne);
+  Value C_new_act = createActivationByName(b, activations.h, newCellState.C);
+  newCellState.H = b.create<AtenMulTensorOp>(hTy, o_act, C_new_act);
+  return newCellState;
+}
+
+struct LstmLayerOutput {
+  Value Y;
+  Value Y_h;
+  Value Y_c;
+};
+
+/**
+ * @brief This function implements the LSTM (Long Short-Term Memory) layer
+ * operation.
+ *
+ * The core computation is performed in a loop that iterates over the sequence
+ * length. In each iteration, it selects the corresponding input, computes the
+ * new hidden state and cell state using the lstm_cell function, and updates the
+ * output tensor.
+ *
+ * @return A struct containing the hidden state historty, final hidden state,
+ * and final cell state.
+ */
+LstmLayerOutput lstm_layer(ImplicitLocOpBuilder &b, Value X, Value initial_h,
+                           Value initial_c, LstmWeights weights,
+                           LstmActivations activations) {
+
+  Location loc = b.getLoc();
+
+  auto xTy = cast<ValueTensorType>(X.getType());
+  auto hTy = cast<ValueTensorType>(initial_h.getType());
+  // these names are snake_case for consistency with onnx.LSTM documentation
+  int64_t seq_len = xTy.getSizes()[0];
+  int64_t batch_size = xTy.getSizes()[1];
+  int64_t input_size = xTy.getSizes()[2];
+  int64_t hidden_size = hTy.getSizes()[1];
+
+  auto cTy = hTy;
+
+  auto intType = b.getType<IntType>();
+
+  Value cstNone = b.create<ConstantNoneOp>();
+  Value cstZero = b.create<ConstantIntOp>(intType, b.getI64IntegerAttr(0));
+  Value cstOne = b.create<ConstantIntOp>(intType, b.getI64IntegerAttr(1));
+  Value cstSeqLen =
+      b.create<ConstantIntOp>(intType, b.getI64IntegerAttr(seq_len));
+  Value cstBatchSize =
+      b.create<ConstantIntOp>(intType, b.getI64IntegerAttr(batch_size));
+  Value cstHiddenSize =
+      b.create<ConstantIntOp>(intType, b.getI64IntegerAttr(hidden_size));
+
+  auto yTy = b.getType<ValueTensorType>(
+      SmallVector<int64_t>{seq_len, batch_size, hidden_size}, hTy.getDtype());
+
+  auto YShapeList = b.create<PrimListConstructOp>(
+      b.getType<ListType>(intType),
+      ValueRange({cstSeqLen, cstBatchSize, cstHiddenSize}));
+
+  int hDtypeInt = (int)getScalarTypeForType(hTy.getDtype());
```

Comment:
use `int64_t` instead, remove the `(int)` cast, it shouldn't be necessary. When you do use casting use `static_cast<int64_t>(...)`